src/runtime/decimal.py

```diff
@@ -0,0 +1,26 @@
+from ltypes import i8, i16, i32, f32, f64, ccall
+
+@overload
+def Decimal(x: i64) -> i64:
+    return x
```

Comment:
```
>>> from decimal import *
>>> Decimal(1)
Decimal('1')
>>> type(Decimal(1))
<class 'decimal.Decimal'>
```
Let's start off with something like the following,

```python
@dataclass
class Decimal:
    value: str
```

Reply:
We can remove `@dataclass` from `Decimal` in future. For now using them should be okay.

Reply:
cc: @certik

Reply:
I wonder if we should just implement these the right way, that is, they have to be implemented as a class, that implements some methods that you can then call. So we have to first implement classes in LPython.

Reply:
Will it have overloading too? Something like:

```
@dataclass
class Decimal:

    @overload
    def Decimal(x: i64) -> i64:
        return x
    @overload
    def Decimal(x: i32) -> i32:
        return x

    @overload
    def Decimal(x: f32) -> f64:
        return float(x)
    @overload
    def Decimal(x: f64) -> f64:
        return float(x)
    
```

Reply:
Also please explain why I am facing this particular error, what should be necessarily done differently?

Reply:
I think we can support overloading of methods, but as a second iteration, not first (it might be a bit involved, but doable).

Why don't you open up an issue for what is needed. I think what is needed is support for this:
```
class Decimal:
    def f(x: i64) -> i64:
        return x

def g() -> Decimal:
    return Decimal()

x: Decimal = g()
print(x.f())
```

And probably more, but the above is a start. Can you please open up an issue for this? Then let's implement it in LPython. Then you'll abe able to use it, and we can then open up issues for any missing features, etc.